lmdeploy/tokenizer.py

```diff
@@ -421,9 +422,9 @@ def encode(self, s: str, add_bos: bool = True, add_special_tokens: bool = True,
         """
         encoded = self.model.encode(s, add_bos, add_special_tokens, **kwargs)
         if encoded[:2] == [self.bos_token_id] * 2:
-            get_logger('lmdeploy').warn(f'Detected duplicate bos token {self.bos_token_id} in prompt, '
-                                        'this will likely reduce response quality, one of them will be'
-                                        'removed')
+            self.logger.warn(f'Detected duplicate bos token {self.bos_token_id} in prompt, '
+                             'this will likely reduce response quality, one of them will be'
+                             'removed')
```

Comment:
The 'warn' method is deprecated, use 'warning' instead

Reply:
fixed